app/assets/javascripts/application.js

```diff
@@ -258,11 +258,18 @@ function initializeAutocomplete(selector, entities) {
             },
             select: function(event, ui) {
                 var terms = split(this.value);
-                // remove the current input
-                terms.pop();
-                // add the selected item
-                terms.push(ui.item.value);
-                terms.push("");
+                // check if user has added more that 20 genes, in which case alert and remove the last term
```

Comment:
looks like terms.pop() could go above the conditional if the comparator becomes a ">=". It's not obvious to me why there is an "extra" pop (result not saved) and push (blank input), which may be just my own lack of context.

Reply:
Oh yeah - I don't know why I didn't think of that.  I just need to adjust the math on the `if` statement.